OY population in King County by race and gender

In [1]:
import psycopg2
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
DBNAME = "opportunity_youth"
conn = psycopg2.connect(dbname=DBNAME)

In [2]:
WA_State_Youth = pd.read_sql("""
            SELECT p.puma, rtrim(puma_name), agep, 
            CASE WHEN p.sex = '1' THEN 'male'
                 ELSE 'female'
                 END as gender,
            CASE WHEN p.racasn = '1' THEN 'asian'
                 WHEN p.racblk = '1' THEN 'black or African American'
                 WHEN p.racwht = '1' THEN 'white'
                 WHEN p.hisp <> '1' THEN 'hispanic' 
                 END as race_or_ethnicity,
            CASE WHEN p.sch = '1' THEN 'Has not attended in last 3 months'
                 WHEN p.sch IN ('2','3') THEN 'public or private school or college and homeschool'
                 END as School,
            CASE WHEN p.schl in ('16','17') THEN 'HS diploma or GED'
                 WHEN p.schl in ('18','19','20','21','22','23','24') THEN 'post HS education'
                 ELSE 'No HS or GED'
                 END as Education_attainment,
            CASE WHEN p.esr in ('1','2') THEN 'employed'
                 WHEN p.esr in ('3','6') THEN 'Unemployed/not in labor force'
                 WHEN p.esr IN ('4','5') THEN 'Armed Forces'
                 END as Employment_status,
            CASE WHEN p.dis = '1' THEN 'reported disability'
                 ELSE 'no reported disability'
                 END as disability_status,
            CASE WHEN p.ddrs = '1' THEN 'Self-care difficulty'
                 WHEN p.dear = '1' THEN 'Hearing difficulty'
                 WHEN p.deye = '1' THEN 'Vision difficulty'
                 WHEN p.dphy = '1' THEN 'Ambulatory difficulty'
                 WHEN p.drem = '1' THEN 'Cognitive difficulty'
                 ELSE 'None reported' 
                 END as disability_type,
            CASE WHEN p.cit IN ('1','2','3','4') THEN 'US Citizen'
                 WHEN p.cit = '5' THEN 'Not US Citizen' 
                 END as Citizenship,
            CASE WHEN p.eng IN ('3','4') THEN 'Poor of No English language'
                 ELSE 'English Speaker'
                 END as English_language,
            CASE WHEN p.wkl = '1' THEN 'employed in the last year'
                 WHEN p.wkl in ('2','3') THEN 'not employed in the last year' 
                 END AS last_employed,
            CASE WHEN p.esp in ('1','2','3','5','6','7','8') THEN 'One or both parents in labor force'
                 WHEN p.esp = '4' THEN 'Neither parent in labor force'
                 END as Parent_employment,
            CASE WHEN p.mar = '1' THEN 'married'
                 WHEN p.mar = '5' THEN 'never married'
                 ELSE 'divorced, widowed, or separated' 
                 END as marital_status,
            CASE WHEN p.fer = '1' THEN 'yes' 
                 END AS Child_born_las_year,
            CASE WHEN p.paoc in ('1','2','3') THEN 'own child(ren)' 
                 ELSE 'no own children' 
                 END as Presence_of_children, 
            CASE WHEN p.mig = '1' THEN 'lived here last year' 
                 ELSE 'moved here last year' 
                 END AS Moved,
            CASE WHEN p.hicov = '1' THEN 'yes' 
                 ELSE 'no' 
                 END as Has_Health_Insurance,
            pap as Public_Assistance_Income_past_year,        
            pwgtp as Weight
            FROM pums_2018 p
            JOIN puma_names_2010 n on p.puma = n.puma
            WHERE state_name = 'Washington'
            --AND p.puma IN ('11610','11611', '11612','11613','11614','11615')
            --AND p.sch = '1' 
            --AND p.esr IN ('3','6')
            AND agep BETWEEN 16 AND 24
            GROUP BY p.puma,n.puma_name,agep,p.sex,p.racasn,p.racblk,p.racwht,p.hisp,p.sch,p.schl,
                     p.esr,p.dis,p.ddrs,p.dear,p.deye,p.dphy,p.drem,p.cit,p.eng,p.wkl,p.esp,
                     p.mar,p.fer,p.paoc,p.mig,p.hicov,p.pap,p.pwgtp
            ORDER BY school, agep DESC""", conn)
WA_State_Youth

puma                                              rtrim  agep  gender  \
0      10400   Stevens, Okanogan, Pend Oreille & Ferry Counties  24.0  female   
1      10400   Stevens, Okanogan, Pend Oreille & Ferry Counties  24.0  female   
2      10400   Stevens, Okanogan, Pend Oreille & Ferry Counties  24.0  female   
3      10400   Stevens, Okanogan, Pend Oreille & Ferry Counties  24.0  female   
4      10400   Stevens, Okanogan, Pend Oreille & Ferry Counties  24.0  female   
...      ...                                                ...   ...     ...   
37750  11702  Snohomish County (West Central)--Mukilteo & Ev...  16.0    male   
37751  11702  Snohomish County (West Central)--Mukilteo & Ev...  16.0    male   
37752  11702  Snohomish County (West Central)--Mukilteo & Ev...  16.0    male   
37753  11702  Snohomish County (West Central)--Mukilteo & Ev...  16.0    male   
37754  11702  Snohomish County (West Central)--Mukilteo & Ev...  16.0    male   

      race_or_ethnicity                                             school  \
0                 white                  Has not attended in last 3 months   
1                 white                  Has not attended in last 3 months   
2                 white                  Has not attended in last 3 months   
3                 white                  Has not attended in last 3 months   
4                 white                  Has not attended in last 3 months   
...                 ...                                                ...   
37750             asian  public or private school or college and homesc...   
37751             asian  public or private school or college and homesc...   
37752             asian  public or private school or college and homesc...   
37753             asian  public or private school or college and homesc...   
37754             asian  public or private school or college and homesc...   

      education_attainment              employment_status  \
0        post HS education  Unemployed/not in labor force   
1        post HS education                       employed   
2        post HS education                       employed   
3        post HS education                       employed   
4        post HS education                       employed   
...                    ...                            ...   
37750         No HS or GED  Unemployed/not in labor force   
37751         No HS or GED  Unemployed/not in labor force   
37752         No HS or GED  Unemployed/not in labor force   
37753         No HS or GED  Unemployed/not in labor force   
37754         No HS or GED  Unemployed/not in labor force   

            disability_status       disability_type  ...  \
0         reported disability  Self-care difficulty  ...   
1      no reported disability         None reported  ...   
2      no reported disability         None reported  ...   
3      no reported disability         None reported  ...   
4      no reported disability         None reported  ...   
...                       ...                   ...  ...   
37750  no reported disability         None reported  ...   
37751  no reported disability         None reported  ...   
37752  no reported disability         None reported  ...   
37753  no reported disability         None reported  ...   
37754  no reported disability         None reported  ...   

                  english_language                  last_employed  \
0                  English Speaker  not employed in the last year   
1                  English Speaker      employed in the last year   
2                  English Speaker      employed in the last year   
3                  English Speaker      employed in the last year   
4                  English Speaker      employed in the last year   
...                            ...                            ...   
37750              English Speaker  not employed in the last year   
37751              English Speaker  not employed in the last year   
37752              Engl

In [3]:
WA_State_Youth['weight'].sum()
State_age_percent_18 = WA_State_Youth.groupby('agep')['weight'].agg(sum)/831436*100
State_race_percent_18 = WA_State_Youth.groupby('race_or_ethnicity')['weight'].agg(sum)/831436*100
State_gender_percent_18 = WA_State_Youth.groupby('gender')['weight'].agg(sum)/831436*100
State_school_percent_18 = WA_State_Youth.groupby('school')['weight'].agg(sum)/831436*100
State_Education_attainment_percent_18 = WA_State_Youth.groupby('education_attainment')['weight'].agg(sum)/831436*100
State_employment_status_percent_18 = WA_State_Youth.groupby('employment_status')['weight'].agg(sum)/831436*100
State_disability_status_percent_18 = WA_State_Youth.groupby('disability_status')['weight'].agg(sum)/831436*100
State_disability_type_percent_18 = WA_State_Youth.groupby('disability_type')['weight'].agg(sum)/831436*100
State_citizenship_percent_18 = WA_State_Youth.groupby('citizenship')['weight'].agg(sum)/831436*100
State_english_language_percent_18 = WA_State_Youth.groupby('english_language')['weight'].agg(sum)/831436*100
State_last_employed_percent_18 = WA_State_Youth.groupby('last_employed')['weight'].agg(sum)/831436*100
State_parent_employment_percent_18 = WA_State_Youth.groupby('parent_employment')['weight'].agg(sum)/831436*100
State_marital_status_percent_18 = WA_State_Youth.groupby('marital_status')['weight'].agg(sum)/831436*100
State_child_born_las_year_percent_18 = WA_State_Youth.groupby('child_born_las_year')['weight'].agg(sum)/831436*100
State_presence_of_children_percent_18 = WA_State_Youth.groupby('presence_of_children')['weight'].agg(sum)/831436*100
State_moved_percent_18 = WA_State_Youth.groupby('moved')['weight'].agg(sum)/831436*100
State_has_health_insurance_percent_18 = WA_State_Youth.groupby('has_health_insurance')['weight'].agg(sum)/831436*100
State_public_assistance_income_past_year_percent_18 = WA_State_Youth.groupby('public_assistance_income_past_year')['weight'].agg(sum)/831436*100

#State_public_assistance_income_past_year_percent_18

In [4]:
King_County_Youth = pd.read_sql("""
            SELECT p.puma, rtrim(puma_name), agep, 
            CASE WHEN p.sex = '1' THEN 'male'
                 ELSE 'female'
                 END as gender,
            CASE WHEN p.racasn = '1' THEN 'asian'
                 WHEN p.racblk = '1' THEN 'black or African American'
                 WHEN p.racwht = '1' THEN 'white'
                 WHEN p.hisp <> '1' THEN 'hispanic' 
                 END as race_or_ethnicity,
            CASE WHEN p.sch = '1' THEN 'Has not attended in last 3 months'
                 WHEN p.sch IN ('2','3') THEN 'public or private school or college and homeschool'
                 END as School,
            CASE WHEN p.schl in ('16','17') THEN 'HS diploma or GED'
                 WHEN p.schl in ('18','19','20','21','22','23','24') THEN 'post HS education'
                 ELSE 'No HS or GED'
                 END as Education_attainment,
            CASE WHEN p.esr in ('1','2') THEN 'employed'
                 WHEN p.esr in ('3','6') THEN 'Unemployed/not in labor force'
                 WHEN p.esr IN ('4','5') THEN 'Armed Forces'
                 END as Employment_status,
            CASE WHEN p.dis = '1' THEN 'reported disability'
                 ELSE 'no reported disability'
                 END as disability_status,
            CASE WHEN p.ddrs = '1' THEN 'Self-care difficulty'
                 WHEN p.dear = '1' THEN 'Hearing difficulty'
                 WHEN p.deye = '1' THEN 'Vision difficulty'
                 WHEN p.dphy = '1' THEN 'Ambulatory difficulty'
                 WHEN p.drem = '1' THEN 'Cognitive difficulty'
                 ELSE 'None reported' 
                 END as disability_type,
            CASE WHEN p.cit IN ('1','2','3','4') THEN 'US Citizen'
                 WHEN p.cit = '5' THEN 'Not US Citizen' 
                 END as Citizenship,
            CASE WHEN p.eng IN ('3','4') THEN 'Poor or No English language'
                 ELSE 'English Speaker'
                 END as English_language,
            CASE WHEN p.wkl = '1' THEN 'employed in the last year'
                 WHEN p.wkl in ('2','3') THEN 'not employed in the last year' 
                 END AS last_employed,
            CASE WHEN p.esp in ('1','2','3','5','6','7','8') THEN 'One or both parents in labor force'
                 WHEN p.esp = '4' THEN 'Neither parent in labor force'
                 END as Parent_employment,
            CASE WHEN p.mar = '1' THEN 'married'
                 WHEN p.mar = '5' THEN 'never married'
                 ELSE 'divorced, widowed, or separated' 
                 END as marital_status,
            CASE WHEN p.fer = '1' THEN 'yes' 
                 END AS Child_born_las_year,
            CASE WHEN p.paoc in ('1','2','3') THEN 'own child(ren)' 
                 ELSE 'no own children' 
                 END as Presence_of_children, 
            CASE WHEN p.mig = '1' THEN 'lived here last year' 
                 ELSE 'moved here last year' 
                 END AS Moved,
            CASE WHEN p.hicov = '1' THEN 'yes' 
                 ELSE 'no' 
                 END as Has_Health_Insurance,
            pap as Public_Assistance_Income_past_year,        
            pwgtp as Weight
            FROM pums_2018 p
            JOIN puma_names_2010 n on p.puma = n.puma
            WHERE state_name = 'Washington'
            AND n.puma_name like ('%King%County%')
            --AND p.sch = '1' 
            --AND p.esr IN ('3','6')
            AND agep BETWEEN 16 AND 24
            GROUP BY p.puma,n.puma_name,agep,p.sex,p.racasn,p.racblk,p.racwht,p.hisp,p.sch,p.schl,
                     p.esr,p.dis,p.ddrs,p.dear,p.deye,p.dphy,p.drem,p.cit,p.eng,p.wkl,p.esp,
                     p.mar,p.fer,p.paoc,p.mig,p.hicov,p.pap,p.pwgtp
            ORDER BY school, agep DESC""", conn)
King_County_Youth

puma                                              rtrim  agep  gender  \
0     11607  King County (Northwest)--Redmond, Kirkland Cit...  24.0  female   
1     11607  King County (Northwest)--Redmond, Kirkland Cit...  24.0  female   
2     11607  King County (Northwest)--Redmond, Kirkland Cit...  24.0  female   
3     11607  King County (Northwest)--Redmond, Kirkland Cit...  24.0  female   
4     11607  King County (Northwest)--Redmond, Kirkland Cit...  24.0  female   
...     ...                                                ...   ...     ...   
6141  11609  King County (Central)--Sammamish, Issaquah, Me...  16.0    male   
6142  11609  King County (Central)--Sammamish, Issaquah, Me...  16.0    male   
6143  11609  King County (Central)--Sammamish, Issaquah, Me...  16.0    male   
6144  11609  King County (Central)--Sammamish, Issaquah, Me...  16.0    male   
6145  11609  King County (Central)--Sammamish, Issaquah, Me...  16.0    male   

              race_or_ethnicity  \
0                         asian   
1                         asian   
2                         asian   
3     black or African American   
4     black or African American   
...                         ...   
6141                      white   
6142                      white   
6143                      white   
6144                      white   
6145                      white   

                                                 school education_attainment  \
0                     Has not attended in last 3 months    post HS education   
1                     Has not attended in last 3 months    post HS education   
2                     Has not attended in last 3 months    post HS education   
3                     Has not attended in last 3 months    post HS education   
4                     Has not attended in last 3 months    HS diploma or GED   
...                                                 ...                  ...   
6141  public or private school or college and homesc...         No HS or GED   
6142  public or private school or college and homesc...         No HS or GED   
6143  public or private school or college and homesc...         No HS or GED   
6144  public or private school or college and homesc...         No HS or GED   
6145  public or private school or college and homesc...         No HS or GED   

                  employment_status       disability_status disability_type  \
0                          employed  no reported disability   None reported   
1                          employed  no reported disability   None reported   
2                          employed  no reported disability   None reported   
3                          employed  no reported disability   None reported   
4     Unemployed/not in labor force  no reported disability   None reported   
...                             ...                     ...             ...   
6141  Unemployed/not in labor force  no reported disability   None reported   
6142  Unemployed/not in labor force  no reported disability   None reported   
6143  Unemployed/not in labor force  no reported disability   None reported   
6144  Unemployed/not in labor force  no reported disability   None reported   
6145  Unemployed/not in labor force  no reported disability   None reported   

      ... english_language                  last_employed  \
0     ...  English Speaker      employed in the last year   
1     ...  English Speaker      employed in the last year   
2     ...  English Speaker      employed in the last year   
3     ...  English Speaker      employed in the last year   
4     ...  English Speaker      employed in the last year   
...   ...              ...                            ...   
6141  ...  English Speaker  not employed in the last year   
6142  ...  English Speaker  not employed in the last year   
6143  ...  English Speaker  not employed in the last year   
6144  ...  English Speaker  not employed in the last year   
6145  ...  English Speaker  not employed in 

In [5]:
County_age_percent_18 = King_County_Youth.groupby('agep')['weight'].agg(sum)/141751*100
County_race_percent_18 = King_County_Youth.groupby('race_or_ethnicity')['weight'].agg(sum)/141751*100
County_gender_percent_18 = King_County_Youth.groupby('gender')['weight'].agg(sum)/141751*100
County_school_percent_18 = King_County_Youth.groupby('school')['weight'].agg(sum)/141751*100
County_Education_attainment_percent_18 = King_County_Youth.groupby('education_attainment')['weight'].agg(sum)/141751*100
County_employment_status_percent_18 = King_County_Youth.groupby('employment_status')['weight'].agg(sum)/141751*100
County_disability_status_percent_18 = King_County_Youth.groupby('disability_status')['weight'].agg(sum)/141751*100
County_disability_type_percent_18 = King_County_Youth.groupby('disability_type')['weight'].agg(sum)/141751*100
County_citizenship_percent_18 = King_County_Youth.groupby('citizenship')['weight'].agg(sum)/141751*100
County_english_language_percent_18 = King_County_Youth.groupby('english_language')['weight'].agg(sum)/141751*100
County_last_employed_percent_18 = King_County_Youth.groupby('last_employed')['weight'].agg(sum)/141751*100
County_parent_employment_percent_18 = King_County_Youth.groupby('parent_employment')['weight'].agg(sum)/141751*100
County_marital_status_percent_18 = King_County_Youth.groupby('marital_status')['weight'].agg(sum)/141751*100
County_child_born_las_year_percent_18 = King_County_Youth.groupby('child_born_las_year')['weight'].agg(sum)/141751*100
County_presence_of_children_percent_18 = King_County_Youth.groupby('presence_of_children')['weight'].agg(sum)/141751*100
County_moved_percent_18 = King_County_Youth.groupby('moved')['weight'].agg(sum)/141751*100
County_has_health_insurance_percent_18 = King_County_Youth.groupby('has_health_insurance')['weight'].agg(sum)/141751*100
County_public_assistance_income_past_year_percent_18 = King_County_Youth.groupby('public_assistance_income_past_year')['weight'].agg(sum)/141751*100

County_age_percent_18

agep
16.0    13.647875
17.0    13.233769
18.0    11.465880
19.0     8.870484
20.0     9.700813
21.0     9.130094
22.0    10.293402
23.0    11.680341
24.0    11.977341
Name: weight, dtype: float64

In [6]:
SouthKing_Youth = pd.read_sql("""
            SELECT p.puma, rtrim(puma_name), agep, 
            CASE WHEN p.sex = '1' THEN 'male'
                 ELSE 'female'
                 END as gender,
            CASE WHEN p.racasn = '1' THEN 'asian'
                 WHEN p.racblk = '1' THEN 'black or African American'
                 WHEN p.racwht = '1' THEN 'white'
                 WHEN p.hisp <> '1' THEN 'hispanic' 
                 END as race_or_ethnicity,
            CASE WHEN p.sch = '1' THEN 'Has not attended in last 3 months'
                 WHEN p.sch IN ('2','3') THEN 'public or private school or college and homeschool'
                 END as School,
            CASE WHEN p.schl in ('16','17') THEN 'HS diploma or GED'
                 WHEN p.schl in ('18','19','20','21','22','23','24') THEN 'post HS education'
                 ELSE 'No HS or GED'
                 END as Education_attainment,
            CASE WHEN p.esr in ('1','2') THEN 'employed'
                 WHEN p.esr in ('3','6') THEN 'Unemployed/not in labor force'
                 WHEN p.esr IN ('4','5') THEN 'Armed Forces'
                 END as Employment_status,
            CASE WHEN p.dis = '1' THEN 'reported disability'
                 ELSE 'no reported disability'
                 END as disability_status,
            CASE WHEN p.ddrs = '1' THEN 'Self-care difficulty'
                 WHEN p.dear = '1' THEN 'Hearing difficulty'
                 WHEN p.deye = '1' THEN 'Vision difficulty'
                 WHEN p.dphy = '1' THEN 'Ambulatory difficulty'
                 WHEN p.drem = '1' THEN 'Cognitive difficulty'
                 ELSE 'None reported' 
                 END as disability_type,
            CASE WHEN p.cit IN ('1','2','3','4') THEN 'US Citizen'
                 WHEN p.cit = '5' THEN 'Not US Citizen' 
                 END as Citizenship,
            CASE WHEN p.eng IN ('3','4') THEN 'Poor or No English language'
                 ELSE 'English speaker'
                 END as English_language,
            CASE WHEN p.wkl = '1' THEN 'employed in the last year'
                 WHEN p.wkl in ('2','3') THEN 'not employed in the last year' 
                 END AS last_employed,
            CASE WHEN p.esp in ('1','2','3','5','6','7','8') THEN 'One or both parents in labor force'
                 WHEN p.esp = '4' THEN 'Neither parent in labor force'
                 END as Parent_employment,
            CASE WHEN p.mar = '1' THEN 'married'
                 WHEN p.mar = '5' THEN 'never married'
                 ELSE 'divorced, widowed, or separated' 
                 END as marital_status,
            CASE WHEN p.fer = '1' THEN 'yes' 
                 END AS Child_born_las_year,
            CASE WHEN p.paoc in ('1','2','3') THEN 'own child(ren)' 
                 ELSE 'no own children' 
                 END as Presence_of_children, 
            CASE WHEN p.mig = '1' THEN 'lived here last year' 
                 ELSE 'moved here last year' 
                 END AS Moved,
            CASE WHEN p.hicov = '1' THEN 'yes' 
                 ELSE 'no' 
                 END as Has_Health_Insurance,
            pap as Public_Assistance_Income_past_year,        
            pwgtp as Weight
            FROM pums_2018 p
            JOIN puma_names_2010 n on p.puma = n.puma
            WHERE state_name = 'Washington'
            AND p.puma IN ('11610','11611', '11612','11613','11614','11615')
            --AND p.sch = '1' 
            --AND p.esr IN ('3','6')
            AND agep BETWEEN 16 AND 24
            GROUP BY p.puma,n.puma_name,agep,p.sex,p.racasn,p.racblk,p.racwht,p.hisp,p.sch,p.schl,
                     p.esr,p.dis,p.ddrs,p.dear,p.deye,p.dphy,p.drem,p.cit,p.eng,p.wkl,p.esp,
                     p.mar,p.fer,p.paoc,p.mig,p.hicov,p.pap,p.pwgtp
            ORDER BY school, agep DESC""", conn)
SouthKing_Youth

puma                                              rtrim  agep  gender  \
0     11610  King County (Central)--Renton City, Fairwood, ...  24.0    male   
1     11610  King County (Central)--Renton City, Fairwood, ...  24.0    male   
2     11610  King County (Central)--Renton City, Fairwood, ...  24.0    male   
3     11611  King County (West Central)--Burien, SeaTac, Tu...  24.0  female   
4     11611  King County (West Central)--Burien, SeaTac, Tu...  24.0  female   
...     ...                                                ...   ...     ...   
3432  11610  King County (Central)--Renton City, Fairwood, ...  16.0  female   
3433  11610  King County (Central)--Renton City, Fairwood, ...  16.0  female   
3434  11610  King County (Central)--Renton City, Fairwood, ...  16.0  female   
3435  11610  King County (Central)--Renton City, Fairwood, ...  16.0  female   
3436  11610  King County (Central)--Renton City, Fairwood, ...  16.0  female   

              race_or_ethnicity  \
0                         white   
1                         white   
2     black or African American   
3                         white   
4     black or African American   
...                         ...   
3432  black or African American   
3433  black or African American   
3434  black or African American   
3435                      white   
3436                      white   

                                                 school education_attainment  \
0                     Has not attended in last 3 months    HS diploma or GED   
1                     Has not attended in last 3 months    post HS education   
2                     Has not attended in last 3 months    HS diploma or GED   
3                     Has not attended in last 3 months    HS diploma or GED   
4                     Has not attended in last 3 months         No HS or GED   
...                                                 ...                  ...   
3432  public or private school or college and homesc...         No HS or GED   
3433  public or private school or college and homesc...         No HS or GED   
3434  public or private school or college and homesc...         No HS or GED   
3435  public or private school or college and homesc...         No HS or GED   
3436  public or private school or college and homesc...         No HS or GED   

                  employment_status       disability_status  \
0                          employed  no reported disability   
1                          employed     reported disability   
2                          employed  no reported disability   
3     Unemployed/not in labor force  no reported disability   
4     Unemployed/not in labor force  no reported disability   
...                             ...                     ...   
3432  Unemployed/not in labor force  no reported disability   
3433  Unemployed/not in labor force  no reported disability   
3434  Unemployed/not in labor force  no reported disability   
3435  Unemployed/not in labor force  no reported disability   
3436  Unemployed/not in labor force  no reported disability   

            disability_type  ...             english_language  \
0             None reported  ...              English speaker   
1     Ambulatory difficulty  ...  Poor or No English language   
2             None reported  ...              English speaker   
3             None reported  ...              English speaker   
4             None reported  ...              English speaker   
...                     ...  ...                          ...   
3432          None reported  ...              English speaker   
3433          None reported  ...              English speaker   
3434          None reported  ...              English speaker   
3435          None reported  ...              English speaker   
3436          None reported  ...              English speaker   

                      last_employed                   parent_employment  \
0         employed in the last year                

In [7]:
SouthKing_Youth_age_percent_18 = SouthKing_Youth.groupby('agep')['weight'].agg(sum)/84956*100
SouthKing_Youth_race_percent_18 = SouthKing_Youth.groupby('race_or_ethnicity')['weight'].agg(sum)/84956*100
SouthKing_Youth_gender_percent_18 = SouthKing_Youth.groupby('gender')['weight'].agg(sum)/84956*100
SouthKing_Youth_school_percent_18 = SouthKing_Youth.groupby('school')['weight'].agg(sum)/84956*100
SouthKing_Youth_Education_attainment_percent_18 = SouthKing_Youth.groupby('education_attainment')['weight'].agg(sum)/84956*100
SouthKing_Youth_employment_status_percent_18 = SouthKing_Youth.groupby('employment_status')['weight'].agg(sum)/84956*100
SouthKing_Youth_disability_status_percent_18 = SouthKing_Youth.groupby('disability_status')['weight'].agg(sum)/84956*100
SouthKing_Youth_disability_type_percent_18 = SouthKing_Youth.groupby('disability_type')['weight'].agg(sum)/84956*100
SouthKing_Youth_citizenship_percent_18 = SouthKing_Youth.groupby('citizenship')['weight'].agg(sum)/84956*100
SouthKing_Youth_english_language_percent_18 = SouthKing_Youth.groupby('english_language')['weight'].agg(sum)/84956*100
SouthKing_Youth_last_employed_percent_18 = SouthKing_Youth.groupby('last_employed')['weight'].agg(sum)/84956*100
SouthKing_Youth_parent_employment_percent_18 = SouthKing_Youth.groupby('parent_employment')['weight'].agg(sum)/84956*100
SouthKing_Youth_marital_status_percent_18 = SouthKing_Youth.groupby('marital_status')['weight'].agg(sum)/84956*100
SouthKing_Youth_child_born_las_year_percent_18 = SouthKing_Youth.groupby('child_born_las_year')['weight'].agg(sum)/84956*100
SouthKing_Youth_presence_of_children_percent_18 = SouthKing_Youth.groupby('presence_of_children')['weight'].agg(sum)/84956*100
SouthKing_Youth_moved_percent_18 = SouthKing_Youth.groupby('moved')['weight'].agg(sum)/84956*100
SouthKing_Youth_has_health_insurance_percent_18 = SouthKing_Youth.groupby('has_health_insurance')['weight'].agg(sum)/84956*100
SouthKing_Youth_public_assistance_income_past_year_percent_18 = SouthKing_Youth.groupby('public_assistance_income_past_year')['weight'].agg(sum)/84956*100

SouthKing_Youth_age_percent_18

agep
16.0    12.388766
17.0    12.219266
18.0    10.525448
19.0     9.594378
20.0    10.922124
21.0     9.530816
22.0    11.338811
23.0    11.372946
24.0    12.107444
Name: weight, dtype: float64

In [8]:
WA_State_OY = pd.read_sql("""
            SELECT p.puma, rtrim(puma_name), agep, 
            CASE WHEN p.sex = '1' THEN 'male'
                 ELSE 'female'
                 END as gender,
            CASE WHEN p.racasn = '1' THEN 'asian'
                 WHEN p.racblk = '1' THEN 'black or African American'
                 WHEN p.racwht = '1' THEN 'white'
                 WHEN p.hisp <> '1' THEN 'hispanic' 
                 END as race_or_ethnicity,
            CASE WHEN p.sch = '1' THEN 'Has not attended in last 3 months'
                 WHEN p.sch IN ('2','3') THEN 'public or private school or college and homeschool'
                 END as School,
            CASE WHEN p.schl in ('16','17') THEN 'HS diploma or GED'
                 WHEN p.schl in ('18','19','20','21','22','23','24') THEN 'post HS education'
                 ELSE 'No HS or GED'
                 END as Education_attainment,
            CASE WHEN p.esr in ('1','2') THEN 'employed'
                 WHEN p.esr in ('3','6') THEN 'Unemployed/not in labor force'
                 WHEN p.esr IN ('4','5') THEN 'Armed Forces'
                 END as Employment_status,
            CASE WHEN p.dis = '1' THEN 'reported disability'
                 ELSE 'no reported disability'
                 END as disability_status,
            CASE WHEN p.ddrs = '1' THEN 'Self-care difficulty'
                 WHEN p.dear = '1' THEN 'Hearing difficulty'
                 WHEN p.deye = '1' THEN 'Vision difficulty'
                 WHEN p.dphy = '1' THEN 'Ambulatory difficulty'
                 WHEN p.drem = '1' THEN 'Cognitive difficulty'
                 ELSE 'None reported' 
                 END as disability_type,
            CASE WHEN p.cit IN ('1','2','3','4') THEN 'US Citizen'
                 WHEN p.cit = '5' THEN 'Not US Citizen' 
                 END as Citizenship,
            CASE WHEN p.eng IN ('3','4') THEN 'Poor or No English language'
                 ELSE 'English Speaker'
                 END as English_language,
            CASE WHEN p.wkl = '1' THEN 'employed in the last year'
                 WHEN p.wkl in ('2','3') THEN 'not employed in the last year' 
                 END AS last_employed,
            CASE WHEN p.esp in ('1','2','3','5','6','7','8') THEN 'One or both parents in labor force'
                 WHEN p.esp = '4' THEN 'Neither parent in labor force'
                 END as Parent_employment,
            CASE WHEN p.mar = '1' THEN 'married'
                 WHEN p.mar = '5' THEN 'never married'
                 ELSE 'divorced, widowed, or separated' 
                 END as marital_status,
            CASE WHEN p.fer = '1' THEN 'yes' 
                 END AS Child_born_las_year,
            CASE WHEN p.paoc in ('1','2','3') THEN 'own child(ren)' 
                 ELSE 'no own children' 
                 END as Presence_of_children, 
            CASE WHEN p.mig = '1' THEN 'lived here last year' 
                 ELSE 'moved here last year' 
                 END AS Moved,
            CASE WHEN p.hicov = '1' THEN 'yes' 
                 ELSE 'no' 
                 END as Has_Health_Insurance,
            pap as Public_Assistance_Income_past_year,        
            pwgtp as Weight
            FROM pums_2018 p
            JOIN puma_names_2010 n on p.puma = n.puma
            WHERE state_name = 'Washington'
            --AND p.puma IN ('11610','11611', '11612','11613','11614','11615')
            AND p.sch = '1' 
            AND p.esr IN ('3','6')
            AND agep BETWEEN 16 AND 24
            GROUP BY p.puma,n.puma_name,agep,p.sex,p.racasn,p.racblk,p.racwht,p.hisp,p.sch,p.schl,
                     p.esr,p.dis,p.ddrs,p.dear,p.deye,p.dphy,p.drem,p.cit,p.eng,p.wkl,p.esp,
                     p.mar,p.fer,p.paoc,p.mig,p.hicov,p.pap,p.pwgtp
            ORDER BY school, agep DESC""", conn)
WA_State_OY

puma                                              rtrim  agep  gender  \
0     10400   Stevens, Okanogan, Pend Oreille & Ferry Counties  24.0    male   
1     10902  Yakima County (Outer)--Sunnyside & Grandview C...  24.0    male   
2     10902  Yakima County (Outer)--Sunnyside & Grandview C...  24.0    male   
3     10902  Yakima County (Outer)--Sunnyside & Grandview C...  24.0    male   
4     10902  Yakima County (Outer)--Sunnyside & Grandview C...  24.0    male   
...     ...                                                ...   ...     ...   
4292  11706  Snohomish County (North)--Marysville & Arlingt...  16.0    male   
4293  11802  Kitsap County (South)--Bremerton & Port Orchar...  16.0    male   
4294  11802  Kitsap County (South)--Bremerton & Port Orchar...  16.0  female   
4295  11802  Kitsap County (South)--Bremerton & Port Orchar...  16.0  female   
4296  11802  Kitsap County (South)--Bremerton & Port Orchar...  16.0  female   

     race_or_ethnicity                             school  \
0                white  Has not attended in last 3 months   
1             hispanic  Has not attended in last 3 months   
2             hispanic  Has not attended in last 3 months   
3                white  Has not attended in last 3 months   
4                white  Has not attended in last 3 months   
...                ...                                ...   
4292          hispanic  Has not attended in last 3 months   
4293             white  Has not attended in last 3 months   
4294          hispanic  Has not attended in last 3 months   
4295             white  Has not attended in last 3 months   
4296             white  Has not attended in last 3 months   

     education_attainment              employment_status  \
0            No HS or GED  Unemployed/not in labor force   
1       HS diploma or GED  Unemployed/not in labor force   
2       post HS education  Unemployed/not in labor force   
3            No HS or GED  Unemployed/not in labor force   
4            No HS or GED  Unemployed/not in labor force   
...                   ...                            ...   
4292         No HS or GED  Unemployed/not in labor force   
4293         No HS or GED  Unemployed/not in labor force   
4294         No HS or GED  Unemployed/not in labor force   
4295         No HS or GED  Unemployed/not in labor force   
4296         No HS or GED  Unemployed/not in labor force   

           disability_status        disability_type  ... english_language  \
0     no reported disability          None reported  ...  English Speaker   
1     no reported disability          None reported  ...  English Speaker   
2     no reported disability          None reported  ...  English Speaker   
3        reported disability  Ambulatory difficulty  ...  English Speaker   
4        reported disability  Ambulatory difficulty  ...  English Speaker   
...                      ...                    ...  ...              ...   
4292  no reported disability          None reported  ...  English Speaker   
4293     reported disability   Self-care difficulty  ...  English Speaker   
4294  no reported disability          None reported  ...  English Speaker   
4295     reported disability   Cognitive difficulty  ...  English Speaker   
4296  no reported disability          None reported  ...  English Speaker   

                      last_employed                   parent_employment  \
0     not employed in the last year                                None   
1         employed in the last year                                None   
2         employed in the last year                                None   
3         employed in the last year                                None   
4         employed in the last year                                None   
...                             ...                                 ...   
4292  not employed in the last year                                None   
4293  not employed in the last year  One or both parents in labor f

In [9]:
WA_State_OY_age_percent_18 = WA_State_OY.groupby('agep')['weight'].agg(sum)/98731*100
WA_State_OY_race_percent_18 = WA_State_OY.groupby('race_or_ethnicity')['weight'].agg(sum)/98731*100
WA_State_OY_gender_percent_18 = WA_State_OY.groupby('gender')['weight'].agg(sum)/987316*100
WA_State_OY_school_percent_18 = WA_State_OY.groupby('school')['weight'].agg(sum)/98731*100
WA_State_OY_Education_attainment_percent_18 = WA_State_OY.groupby('education_attainment')['weight'].agg(sum)/98731*100
WA_State_OY_employment_status_percent_18 = WA_State_OY.groupby('employment_status')['weight'].agg(sum)/98731*100
WA_State_OY_disability_status_percent_18 = WA_State_OY.groupby('disability_status')['weight'].agg(sum)/98731*100
WA_State_OY_disability_type_percent_18 = WA_State_OY.groupby('disability_type')['weight'].agg(sum)/98731*100
WA_State_OY_citizenship_percent_18 = WA_State_OY.groupby('citizenship')['weight'].agg(sum)/98731*100
WA_State_OY_english_language_percent_18 = WA_State_OY.groupby('english_language')['weight'].agg(sum)/98731*100
WA_State_OY_last_employed_percent_18 = WA_State_OY.groupby('last_employed')['weight'].agg(sum)/98731*100
WA_State_OY_parent_employment_percent_18 = WA_State_OY.groupby('parent_employment')['weight'].agg(sum)/98731*100
WA_State_OY_marital_status_percent_18 = WA_State_OY.groupby('marital_status')['weight'].agg(sum)/98731*100
WA_State_OY_child_born_las_year_percent_18 = WA_State_OY.groupby('child_born_las_year')['weight'].agg(sum)/98731*100
WA_State_OY_presence_of_children_percent_18 = WA_State_OY.groupby('presence_of_children')['weight'].agg(sum)/98731*100
WA_State_OY_moved_percent_18 = WA_State_OY.groupby('moved')['weight'].agg(sum)/98731*100
WA_State_OY_has_health_insurance_percent_18 = WA_State_OY.groupby('has_health_insurance')['weight'].agg(sum)/98731*100
WA_State_OY_public_assistance_income_past_year_percent_18 = WA_State_OY.groupby('public_assistance_income_past_year')['weight'].agg(sum)/98731*100

WA_State_OY_age_percent_18

agep
16.0     2.236380
17.0     3.229989
18.0     7.821252
19.0    11.259888
20.0    13.825445
21.0    14.306550
22.0    14.873748
23.0    16.838683
24.0    15.608066
Name: weight, dtype: float64

In [10]:
King_County_OY = pd.read_sql("""
            SELECT p.puma, rtrim(puma_name), agep, 
            CASE WHEN p.sex = '1' THEN 'male'
                 ELSE 'female'
                 END as gender,
            CASE WHEN p.racasn = '1' THEN 'asian'
                 WHEN p.racblk = '1' THEN 'black or African American'
                 WHEN p.racwht = '1' THEN 'white'
                 WHEN p.hisp <> '1' THEN 'hispanic' 
                 END as race_or_ethnicity,
            CASE WHEN p.sch = '1' THEN 'Has not attended in last 3 months'
                 WHEN p.sch IN ('2','3') THEN 'public or private school or college and homeschool'
                 END as School,
            CASE WHEN p.schl in ('16','17') THEN 'HS diploma or GED'
                 WHEN p.schl in ('18','19','20','21','22','23','24') THEN 'post HS education'
                 ELSE 'No HS or GED'
                 END as Education_attainment,
            CASE WHEN p.esr in ('1','2') THEN 'employed'
                 WHEN p.esr in ('3','6') THEN 'Unemployed/not in labor force'
                 WHEN p.esr IN ('4','5') THEN 'Armed Forces'
                 END as Employment_status,
            CASE WHEN p.dis = '1' THEN 'reported disability'
                 ELSE 'no reported disability'
                 END as disability_status,
            CASE WHEN p.ddrs = '1' THEN 'Self-care difficulty'
                 WHEN p.dear = '1' THEN 'Hearing difficulty'
                 WHEN p.deye = '1' THEN 'Vision difficulty'
                 WHEN p.dphy = '1' THEN 'Ambulatory difficulty'
                 WHEN p.drem = '1' THEN 'Cognitive difficulty'
                 ELSE 'None reported' 
                 END as disability_type,
            CASE WHEN p.cit IN ('1','2','3','4') THEN 'US Citizen'
                 WHEN p.cit = '5' THEN 'Not US Citizen' 
                 END as Citizenship,
            CASE WHEN p.eng IN ('3','4') THEN 'Poor or No English language'
                 ELSE 'English speaker'
                 END as English_language,
            CASE WHEN p.wkl = '1' THEN 'employed in the last year'
                 WHEN p.wkl in ('2','3') THEN 'not employed in the last year' 
                 END AS last_employed,
            CASE WHEN p.esp in ('1','2','3','5','6','7','8') THEN 'One or both parents in labor force'
                 WHEN p.esp = '4' THEN 'Neither parent in labor force'
                 END as Parent_employment,
            CASE WHEN p.mar = '1' THEN 'married'
                 WHEN p.mar = '5' THEN 'never married'
                 ELSE 'divorced, widowed, or separated' 
                 END as marital_status,
            CASE WHEN p.fer = '1' THEN 'yes' 
                 END AS Child_born_las_year,
            CASE WHEN p.paoc in ('1','2','3') THEN 'own child(ren)' 
                 ELSE 'no own children' 
                 END as Presence_of_children, 
            CASE WHEN p.mig = '1' THEN 'lived here last year' 
                 ELSE 'moved here last year' 
                 END AS Moved,
            CASE WHEN p.hicov = '1' THEN 'yes' 
                 ELSE 'no' 
                 END as Has_Health_Insurance,
            pap as Public_Assistance_Income_past_year,        
            pwgtp as Weight
            FROM pums_2018 p
            JOIN puma_names_2010 n on p.puma = n.puma
            WHERE state_name = 'Washington'
            AND n.puma_name like ('%King%County%')
            AND p.sch = '1' 
            AND p.esr IN ('3','6')
            AND agep BETWEEN 16 AND 24
            GROUP BY p.puma,n.puma_name,agep,p.sex,p.racasn,p.racblk,p.racwht,p.hisp,p.sch,p.schl,
                     p.esr,p.dis,p.ddrs,p.dear,p.deye,p.dphy,p.drem,p.cit,p.eng,p.wkl,p.esp,
                     p.mar,p.fer,p.paoc,p.mig,p.hicov,p.pap,p.pwgtp
            ORDER BY school, agep DESC""", conn)
King_County_OY

puma                                              rtrim  agep  gender  \
0    11614    King County (Southwest)--Auburn City & Lakeland  24.0  female   
1    11614    King County (Southwest)--Auburn City & Lakeland  24.0  female   
2    11614    King County (Southwest)--Auburn City & Lakeland  24.0  female   
3    11614    King County (Southwest)--Auburn City & Lakeland  24.0  female   
4    11614    King County (Southwest)--Auburn City & Lakeland  24.0  female   
..     ...                                                ...   ...     ...   
543  11612  King County (Far Southwest)--Federal Way, Des ...  16.0  female   
544  11615  King County (Southeast)--Maple Valley, Covingt...  16.0  female   
545  11614    King County (Southwest)--Auburn City & Lakeland  16.0  female   
546  11611  King County (West Central)--Burien, SeaTac, Tu...  16.0    male   
547  11611  King County (West Central)--Burien, SeaTac, Tu...  16.0    male   

             race_or_ethnicity                             school  \
0                        asian  Has not attended in last 3 months   
1                        asian  Has not attended in last 3 months   
2                        asian  Has not attended in last 3 months   
3    black or African American  Has not attended in last 3 months   
4                        white  Has not attended in last 3 months   
..                         ...                                ...   
543                      white  Has not attended in last 3 months   
544                      white  Has not attended in last 3 months   
545                   hispanic  Has not attended in last 3 months   
546                      white  Has not attended in last 3 months   
547                      white  Has not attended in last 3 months   

    education_attainment              employment_status  \
0      post HS education  Unemployed/not in labor force   
1      post HS education  Unemployed/not in labor force   
2           No HS or GED  Unemployed/not in labor force   
3      post HS education  Unemployed/not in labor force   
4      HS diploma or GED  Unemployed/not in labor force   
..                   ...                            ...   
543         No HS or GED  Unemployed/not in labor force   
544         No HS or GED  Unemployed/not in labor force   
545         No HS or GED  Unemployed/not in labor force   
546         No HS or GED  Unemployed/not in labor force   
547         No HS or GED  Unemployed/not in labor force   

          disability_status disability_type  ... english_language  \
0    no reported disability   None reported  ...  English speaker   
1    no reported disability   None reported  ...  English speaker   
2    no reported disability   None reported  ...  English speaker   
3    no reported disability   None reported  ...  English speaker   
4    no reported disability   None reported  ...  English speaker   
..                      ...             ...  ...              ...   
543  no reported disability   None reported  ...  English speaker   
544     reported disability   None reported  ...  English speaker   
545  no reported disability   None reported  ...  English speaker   
546  no reported disability   None reported  ...  English speaker   
547  no reported disability   None reported  ...  English speaker   

                     last_employed                   parent_employment  \
0        employed in the last year                                None   
1        employed in the last year                                None   
2    not employed in the last year                                None   
3        employed in the last year                                None   
4    not employed in the last year                                None   
..                             ...                                 ...   
543  not employed in the last year  One or both parents in labor force   
544  not employed in the last year  One or both parents in labor force   
545  not employed in the

In [11]:
King_County_OY_age_percent_18 = King_County_OY.groupby('agep')['weight'].agg(sum)/14061*100
King_County_OY_race_percent_18 = King_County_OY.groupby('race_or_ethnicity')['weight'].agg(sum)/14061*100
King_County_OY_gender_percent_18 = King_County_OY.groupby('gender')['weight'].agg(sum)/14061*100
King_County_OY_school_percent_18 = King_County_OY.groupby('school')['weight'].agg(sum)/14061*100
King_County_OY_Education_attainment_percent_18 = King_County_OY.groupby('education_attainment')['weight'].agg(sum)/14061*100
King_County_OY_employment_status_percent_18 = King_County_OY.groupby('employment_status')['weight'].agg(sum)/14061*100
King_County_OY_disability_status_percent_18 = King_County_OY.groupby('disability_status')['weight'].agg(sum)/14061*100
King_County_OY_disability_type_percent_18 = King_County_OY.groupby('disability_type')['weight'].agg(sum)/14061*100
King_County_OY_citizenship_percent_18 = King_County_OY.groupby('citizenship')['weight'].agg(sum)/14061*100
King_County_OY_english_language_percent_18 = King_County_OY.groupby('english_language')['weight'].agg(sum)/14061*100
King_County_OY_last_employed_percent_18 = King_County_OY.groupby('last_employed')['weight'].agg(sum)/14061*100
King_County_OY_parent_employment_percent_18 = King_County_OY.groupby('parent_employment')['weight'].agg(sum)/14061*100
King_County_OY_marital_status_percent_18 = King_County_OY.groupby('marital_status')['weight'].agg(sum)/14061*100
King_County_OY_child_born_las_year_percent_18 = King_County_OY.groupby('child_born_las_year')['weight'].agg(sum)/14061*100
King_County_OY_presence_of_children_percent_18 = King_County_OY.groupby('presence_of_children')['weight'].agg(sum)/14061*100
King_County_OY_moved_percent_18 = King_County_OY.groupby('moved')['weight'].agg(sum)/14061*100
King_County_OY_has_health_insurance_percent_18 = King_County_OY.groupby('has_health_insurance')['weight'].agg(sum)/14061*100

King_County_OY_age_percent_18

agep
16.0     3.228789
17.0     1.849086
18.0     9.359221
19.0    13.434322
20.0    11.258090
21.0    12.004836
22.0    14.991821
23.0    18.675770
24.0    15.198066
Name: weight, dtype: float64

In [12]:
SouthKing_OY = pd.read_sql("""
            SELECT p.puma, rtrim(puma_name), agep, 
            CASE WHEN p.sex = '1' THEN 'male'
                 ELSE 'female'
                 END as gender,
            CASE WHEN p.racasn = '1' THEN 'asian'
                 WHEN p.racblk = '1' THEN 'black or African American'
                 WHEN p.racwht = '1' THEN 'white'
                 WHEN p.hisp <> '1' THEN 'hispanic' 
                 END as race_or_ethnicity,
            CASE WHEN p.sch = '1' THEN 'Has not attended in last 3 months'
                 WHEN p.sch IN ('2','3') THEN 'public or private school or college and homeschool'
                 END as School,
            CASE WHEN p.schl in ('16','17') THEN 'HS diploma or GED'
                 WHEN p.schl in ('18','19','20','21','22','23','24') THEN 'post HS education'
                 ELSE 'No HS or GED'
                 END as Education_attainment,
            CASE WHEN p.esr in ('1','2') THEN 'employed'
                 WHEN p.esr in ('3','6') THEN 'Unemployed/not in labor force'
                 WHEN p.esr IN ('4','5') THEN 'Armed Forces'
                 END as Employment_status,
            CASE WHEN p.dis = '1' THEN 'reported disability'
                 ELSE 'no reported disability'
                 END as disability_status,
            CASE WHEN p.ddrs = '1' THEN 'Self-care difficulty'
                 WHEN p.dear = '1' THEN 'Hearing difficulty'
                 WHEN p.deye = '1' THEN 'Vision difficulty'
                 WHEN p.dphy = '1' THEN 'Ambulatory difficulty'
                 WHEN p.drem = '1' THEN 'Cognitive difficulty'
                 ELSE 'None reported' 
                 END as disability_type,
            CASE WHEN p.cit IN ('1','2','3','4') THEN 'US Citizen'
                 WHEN p.cit = '5' THEN 'Not US Citizen' 
                 END as Citizenship,
            CASE WHEN p.eng IN ('3','4') THEN 'Poor or No English Language'
                 ELSE 'English Speaker'
                 END as English_language,
            CASE WHEN p.wkl = '1' THEN 'employed in the last year'
                 WHEN p.wkl in ('2','3') THEN 'not employed in the last year' 
                 END AS last_employed,
            CASE WHEN p.esp in ('1','2','3','5','6','7','8') THEN 'One or both parents in labor force'
                 WHEN p.esp = '4' THEN 'Neither parent in labor force'
                 END as Parent_employment,
            CASE WHEN p.mar = '1' THEN 'married'
                 WHEN p.mar = '5' THEN 'never married'
                 ELSE 'divorced, widowed, or separated' 
                 END as marital_status,
            CASE WHEN p.fer = '1' THEN 'yes' 
                 END AS Child_born_las_year,
            CASE WHEN p.paoc in ('1','2','3') THEN 'own child(ren)' 
                 ELSE 'no own children' 
                 END as Presence_of_children, 
            CASE WHEN p.mig = '1' THEN 'lived here last year' 
                 ELSE 'moved here last year' 
                 END AS Moved,
            CASE WHEN p.hicov = '1' THEN 'yes' 
                 ELSE 'no' 
                 END as Has_Health_Insurance,
            pap as Public_Assistance_Income_past_year,        
            pwgtp as Weight
            FROM pums_2018 p
            JOIN puma_names_2010 n on p.puma = n.puma
            WHERE state_name = 'Washington'
            AND p.puma IN ('11610','11611', '11612','11613','11614','11615')
            AND p.sch = '1' 
            AND p.esr IN ('3','6')
            AND agep BETWEEN 16 AND 24
            GROUP BY p.puma,n.puma_name,agep,p.sex,p.racasn,p.racblk,p.racwht,p.hisp,p.sch,p.schl,
                     p.esr,p.dis,p.ddrs,p.dear,p.deye,p.dphy,p.drem,p.cit,p.eng,p.wkl,p.esp,
                     p.mar,p.fer,p.paoc,p.mig,p.hicov,p.pap,p.pwgtp
            ORDER BY school, agep DESC""", conn)
SouthKing_OY

puma                                              rtrim  agep  gender  \
0    11614    King County (Southwest)--Auburn City & Lakeland  24.0  female   
1    11614    King County (Southwest)--Auburn City & Lakeland  24.0  female   
2    11614    King County (Southwest)--Auburn City & Lakeland  24.0  female   
3    11614    King County (Southwest)--Auburn City & Lakeland  24.0  female   
4    11614    King County (Southwest)--Auburn City & Lakeland  24.0    male   
..     ...                                                ...   ...     ...   
361  11612  King County (Far Southwest)--Federal Way, Des ...  16.0  female   
362  11614    King County (Southwest)--Auburn City & Lakeland  16.0    male   
363  11614    King County (Southwest)--Auburn City & Lakeland  16.0    male   
364  11615  King County (Southeast)--Maple Valley, Covingt...  16.0  female   
365  11610  King County (Central)--Renton City, Fairwood, ...  16.0    male   

             race_or_ethnicity                             school  \
0                        asian  Has not attended in last 3 months   
1    black or African American  Has not attended in last 3 months   
2                        white  Has not attended in last 3 months   
3                        white  Has not attended in last 3 months   
4                     hispanic  Has not attended in last 3 months   
..                         ...                                ...   
361                      white  Has not attended in last 3 months   
362                      white  Has not attended in last 3 months   
363                      white  Has not attended in last 3 months   
364                      white  Has not attended in last 3 months   
365                      white  Has not attended in last 3 months   

    education_attainment              employment_status  \
0           No HS or GED  Unemployed/not in labor force   
1      post HS education  Unemployed/not in labor force   
2      HS diploma or GED  Unemployed/not in labor force   
3      post HS education  Unemployed/not in labor force   
4           No HS or GED  Unemployed/not in labor force   
..                   ...                            ...   
361         No HS or GED  Unemployed/not in labor force   
362    HS diploma or GED  Unemployed/not in labor force   
363         No HS or GED  Unemployed/not in labor force   
364         No HS or GED  Unemployed/not in labor force   
365         No HS or GED  Unemployed/not in labor force   

          disability_status disability_type  ... english_language  \
0    no reported disability   None reported  ...  English Speaker   
1    no reported disability   None reported  ...  English Speaker   
2    no reported disability   None reported  ...  English Speaker   
3    no reported disability   None reported  ...  English Speaker   
4    no reported disability   None reported  ...  English Speaker   
..                      ...             ...  ...              ...   
361  no reported disability   None reported  ...  English Speaker   
362  no reported disability   None reported  ...  English Speaker   
363  no reported disability   None reported  ...  English Speaker   
364     reported disability   None reported  ...  English Speaker   
365  no reported disability   None reported  ...  English Speaker   

                     last_employed                   parent_employment  \
0    not employed in the last year                                None   
1        employed in the last year                                None   
2    not employed in the last year                                None   
3        employed in the last year                                None   
4    not employed in the last year                                None   
..                             ...                                 ...   
361  not employed in the last year  One or both parents in labor force   
362  not employed in the last year  One or both parents in labor force   
363  not employed in the

In [34]:
SouthKing_OY_age_percent_18 = SouthKing_OY.groupby('agep')['weight'].agg(sum)/9806*100
SouthKing_OY_race_percent_18 = SouthKing_OY.groupby('race_or_ethnicity')['weight'].agg(sum)/9806*100
SouthKing_OY_gender_percent_18 = SouthKing_OY.groupby('gender')['weight'].agg(sum)/9806*100
SouthKing_OY_school_percent_18 = SouthKing_OY.groupby('school')['weight'].agg(sum)/9806*100
SouthKing_OY_Education_attainment_percent_18 = SouthKing_OY.groupby('education_attainment')['weight'].agg(sum)/9806*100
SouthKing_OY_employment_status_percent_18 = SouthKing_OY.groupby('employment_status')['weight'].agg(sum)/9806*100
SouthKing_OY_disability_status_percent_18 = SouthKing_OY.groupby('disability_status')['weight'].agg(sum)/9806*100
SouthKing_OY_disability_type_percent_18 = SouthKing_OY.groupby('disability_type')['weight'].agg(sum)/9806*100
SouthKing_OY_citizenship_percent_18 = SouthKing_OY.groupby('citizenship')['weight'].agg(sum)/9806*100
SouthKing_OY_english_language_percent_18 = SouthKing_OY.groupby('english_language')['weight'].agg(sum)/9806*100
SouthKing_OY_last_employed_percent_18 = SouthKing_OY.groupby('last_employed')['weight'].agg(sum)/9806*100
SouthKing_OY_parent_employment_percent_18 = SouthKing_OY.groupby('parent_employment')['weight'].agg(sum)/9806*100
SouthKing_OY_marital_status_percent_18 = SouthKing_OY.groupby('marital_status')['weight'].agg(sum)/9806*100
SouthKing_OY_child_born_las_year_percent_18 = SouthKing_OY.groupby('child_born_las_year')['weight'].agg(sum)/9806*100
SouthKing_OY_presence_of_children_percent_18 = SouthKing_OY.groupby('presence_of_children')['weight'].agg(sum)/9806*100
SouthKing_OY_moved_percent_18 = SouthKing_OY.groupby('moved')['weight'].agg(sum)/9806*100
SouthKing_OY_has_health_insurance_percent_18 = SouthKing_OY.groupby('has_health_insurance')['weight'].agg(sum)/9806*100
SouthKing_OY_public_assistance_income_past_year_percent_18 = SouthKing_OY.groupby('public_assistance_income_past_year')['weight'].agg(sum)/9806*100
SouthKing_OY_puma_18 = SouthKing_OY.groupby('puma')['weight'].agg(sum)/9806*100
SouthKing_OY_puma_18

puma
11610    16.234958
11611    16.031001
11612    21.425658
11613    19.671630
11614    15.643484
11615    10.993269
Name: weight, dtype: float64

Total 16-24 population in Washington State = 831,436
Total 16-24 population in King County, WA = 141,751
Total 16-24 population in South King County, WA = 84,956
OY population in Washington State = 98,731 (11.8%)
OY population in King County, WA = 14,061 (9.9%)
OY population in South King County, WA = 9806 (11.5%)

The South King County OY population is roughly in line with what we see state-wide, the overall King County OY population is slightly smaller percentage-wise than the state.  If we break it down a little further we see that the area outside of South King County (we will refer to it as North King) has a total 16-24 population of 56,795 and an OY ppulation of 4,255 or 7.5%, so the differences we see between the county and the rest of the state are accounted for by the lower rate of OY in the North King region

In South King County, the OY population has a much greater likelihood giving birth in the last year and more likely to be married than King county overall 
OY population = 9.3% gave birth in the last 12 months
Overall = 2.4% gave birth in the last 12 months
OY population = 16.2 percent are married
Overall = 7.4% ae married

In [14]:
SouthKing_OY_child_born_las_year_percent_18

child_born_las_year
yes    9.331022
Name: weight, dtype: float64

In [15]:
SouthKing_Youth_child_born_las_year_percent_18

child_born_las_year
yes    2.393003
Name: weight, dtype: float64

In [16]:
SouthKing_OY_marital_status_percent_18

marital_status
divorced, widowed, or separated     0.397716
married                            16.224760
never married                      83.377524
Name: weight, dtype: float64

In [17]:
SouthKing_Youth_marital_status_percent_18

marital_status
divorced, widowed, or separated     0.573238
married                             7.387353
never married                      92.039409
Name: weight, dtype: float64

are slightly more likely to not be US Citizens
OY Citizenship status = 17.6% Not US Citizen
Overall = 14.9% Not US Citizen

In [18]:
SouthKing_OY_citizenship_percent_18

citizenship
Not US Citizen    17.581073
US Citizen        82.418927
Name: weight, dtype: float64

In [19]:
SouthKing_Youth_citizenship_percent_18

citizenship
Not US Citizen    14.895946
US Citizen        85.104054
Name: weight, dtype: float64

are more likely to have a reported disability
OY reported disability = 15.4%
Overall reported disability = 6.3%

In [20]:
SouthKing_OY_disability_status_percent_18

disability_status
no reported disability    84.060779
reported disability       15.939221
Name: weight, dtype: float64

In [21]:
SouthKing_Youth_disability_status_percent_18

disability_status
no reported disability    93.702623
reported disability        6.297377
Name: weight, dtype: float64

far more likely to have poor english language skills
OY english = 7.1%
Overall = 3.3%

In [22]:
SouthKing_OY_english_language_percent_18

english_language
English Speaker                92.892107
Poor or No English Language     7.107893
Name: weight, dtype: float64

In [23]:
SouthKing_Youth_english_language_percent_18

english_language
English speaker                96.691228
Poor or No English language     3.308772
Name: weight, dtype: float64

more likely to be uninsured
OY health insurance = 21.8% uninsured
Overall health insurance = 11.5% uninsured

In [24]:
SouthKing_OY_has_health_insurance_percent_18

has_health_insurance
no     21.802978
yes    78.197022
Name: weight, dtype: float64

In [25]:
SouthKing_Youth_has_health_insurance_percent_18

has_health_insurance
no     11.494185
yes    88.505815
Name: weight, dtype: float64

slightly more likely to have moved to the area in the last year
OY = 34.2% moved to the area in the last year
Overall = 24.2% moved to the area in the last year

In [26]:
SouthKing_OY_moved_percent_18

moved
lived here last year    65.837243
moved here last year    34.162757
Name: weight, dtype: float64

In [27]:
SouthKing_Youth_moved_percent_18

moved
lived here last year    75.805123
moved here last year    24.194877
Name: weight, dtype: float64

more likely to be in homes where neither parent is employed
OY = 4.7% in homes where one or both parents are in labor force
Overall = 22.13%

In [28]:
SouthKing_OY_parent_employment_percent_18

parent_employment
One or both parents in labor force    4.701203
Name: weight, dtype: float64

In [29]:
SouthKing_Youth_parent_employment_percent_18

parent_employment
Neither parent in labor force          0.173031
One or both parents in labor force    22.133810
Name: weight, dtype: float64

The OY population in South King County is less likely to have any type of post HS education
The differences seen in HS Diploma/GED is explained by the 16-18 year age group still attending school, we would not expect this group to have a significant percentage of HS/GED attainment.

In [30]:
SouthKing_OY_Education_attainment_percent_18


education_attainment
HS diploma or GED    50.999388
No HS or GED         32.102794
post HS education    16.897818
Name: weight, dtype: float64

In [31]:
SouthKing_Youth_Education_attainment_percent_18

education_attainment
HS diploma or GED    26.230048
No HS or GED         36.708414
post HS education    37.061538
Name: weight, dtype: float64

In [32]:
SouthKing_OY_last_employed_percent_18

last_employed
employed in the last year        41.831532
not employed in the last year    58.168468
Name: weight, dtype: float64

In [ ]:
SouthKing